traffic_ops/traffic_ops_golang/parameter/parameters.go

```diff
@@ -364,23 +363,28 @@ func CreateParameter(w http.ResponseWriter, r *http.Request) {
 
 	// This code block decides if the request body is a slice of parameters or a single object.
 	var params []tc.ParameterV5
-	switch reflect.TypeOf(data).Kind() {
-	case reflect.Slice:
-		if err := json.Unmarshal(body, &params); err != nil {
-			api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("error unmarshalling slice"), nil)
-			return
+
+	// Try to unmarshal the data as a slice
+	var sliceData []map[string]interface{}
+	if err := json.Unmarshal(body, &sliceData); err == nil {
+		for _, item := range sliceData {
+			param := tc.ParameterV5{
+				ConfigFile: item["configFile"].(string),
+				Name:       item["name"].(string),
+				Secure:     item["secure"].(bool),
+				Value:      item["value"].(string),
+			}
+
+			params = append(params, param)
 		}
-	case reflect.Map:
-		// If it is a single object it is still converted to a slice for code simplicity.
+	} else {
+		// Try to unmarshal the data as a single object
```

Comment:
`sliceData` is unnecessary. Why not just unmarshal directly to `params`?

```go
if err := json.Unmarshal(body, &params); err != nil {
    var param tc.ParameterV5
    if err := json.Unmarshal(body, &param); err != nil {
        api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("error unmarshalling single object"), nil)
        return
    }
    params = append(params, param)
}
```

Reply:
Removed sliceData part